### ISSUE TITLE
[mdc-slider] Pointer's and slider's positions are out of sync in mdc-dialog

### ISSUE THREAD

**Issue by pajats** [material-components/material-components-web#5128]
## Bug report
When parent component is scrollable and scrolled, slider in MdcDialog is "out of sync" exactly the amount the parent component is scrolled.

### Steps to reproduce
1. Go to https://stackblitz.com/edit/angular-mdc-slider-resize-bug-9ivgw4
2. scroll to the left
3. open dialog from 'Open' button
4. try to slide the slider

### Actual behavior
if I click on a slider track, the slider pin moves away from the point I clicked

### Expected behavior
if I click on a slider track, the slider pin moves to the point I clicked 

### Screenshots
The exmaple with the erroneous behaviour is in https://stackblitz.com/edit/angular-mdc-slider-resize-bug-9ivgw4

### Your Environment:

| Software         | Version(s) |
| ---------------- | ---------- |
| MDC Web          | 3.2.1
| Browser          | Chrome 77.0.3865.90
| Operating System | Ubuntu 18.04.3 LTS

**Comment by bonniezhou** [comment]
You may need to call `layout` on both the dialog and slider upon opening the dialog. See the example [here](https://github.com/material-components/material-components-web/tree/master/packages/mdc-dialog#javascript-instantiation).

**Comment by CyborgSemon** [comment]
Yeah I'm not 100% sure how typescript works so I can't make an example, but you need to call `.layout()` on the slider **after** the dialog has opened. You can listen for the dialog opening by using `MDCDialog:opened`
Here is an example using the default JS import on the component docs: (Not sure what it would look like in TS, but I'm sure you will be able to convert it 😄)
```js
import {MDCSlider} from '@material/slider';
const slider = new MDCSlider(document.querySelector('.mdc-slider'));

import {MDCDialog} from '@material/dialog';
const dialog = new MDCDialog(document.querySelector('.mdc-dialog'));

dialog.listen('MDCDialog:opened', () => {
    slider.layout();
});
```
Hope this helps!

**Comment by pajats** [comment]
@CyborgSemon, should it be done some how differently as done in the example (https://stackblitz.com/edit/angular-mdc-slider-resize-bug-9ivgw4)?

**Comment by CyborgSemon** [comment]
So, again, I've never used TypeScript before so I'm not 100% sure what I'm doing, But the way that I got it working on your example is by putting this into line 13 on `ExampleDialog.component.ts`:
```ts
setTimeout(()=> {
    this.slider.layout();
}, 100);
```

I'm not sure how to get the first example I provided work work in TypeScript, but that is the proper way to get it working. The reason is because the slider becomes visible**during** the dialog opening animation. So it sets up its sizing based off the original dialog size which is quite small when it opens. To stop the bug from happening, you need to wait for the dialog to finish its opening animation before you can call `.layout()` on the slider.
As the docs said in the link that @bonniezhou provided,

> MDC Dialog makes no assumptions about what will be added to the `mdc-dialog__content` element. Any List, Checkboxes, etc. must also be instantiated. If your dialog contains any layout-sensitive components, you should wait until `MDCDialog:opened` is emitted to instantiate them (or call `layout` on them) so that the dialog's transition finishes first.

So the fix I've just posted will work, but you should try to get the listen version working as if the animation timings ever change, you wont have to update the `setTimeout` function yourself.

**Comment by pajats** [comment]
@CyborgSemon, thanks for your effort, but I must do something wrong as I still can't it working in my example, even then I replaced my line of code with code provided by you. Did you scroll the window where the button resides to the left before opening the dialog?

**Comment by CyborgSemon** [comment]
Nah I didn't.
I might have gotten my timing wrong :P whoops sorry
Try 150 instead of 100?

**Comment by raidokaldma** [comment]
I'm late to the party but I would like to add that it does not seem to be a timing issue. Running it with `setTimeout()` or even with `setInterval()` makes no difference. 

When underlying page is horizontally scrolled then this still happens:
![Peek 2019-10-04 08-55](https://user-images.githubusercontent.com/5031837/66184540-2c502700-e685-11e9-8083-292b3cd1f59f.gif)

**Comment by raidokaldma** [comment]
Just to be clear, this issue appears when slider is within an element which has a fixed position. It does not necessarily have to be in a MDCDialog.

So whenever page is horizontally scrolled, this calculation will provide incorrect result:
https://github.com/material-components/material-components-web/blob/master/packages/mdc-slider/foundation.ts#L333

Using `evt.clientX` instead of `evt.pageX` would fix this specific issue.

**Comment by CyborgSemon** [comment]
Thats really odd @raidokaldma. This is what happens to me if I don't call `.layout()` on the slider:

![bug1](https://user-images.githubusercontent.com/26075792/66190318-3ce5b900-e6e8-11e9-9fe2-fef3fc17d0cd.gif)

And this is what happens to me when I call `.layout()` using the `setTimeout()` method. (Still should be listening to the `MDCDialog:opened` event in the first place, but whatever. I don't know TypeScript :P):

![bug2](https://user-images.githubusercontent.com/26075792/66190336-453df400-e6e8-11e9-9469-acd610b6d883.gif)

This is in Chrome 77 on macos. What are you guys running on @raidokaldma @pajats?

**Comment by pajats** [comment]
@CyborgSemon, you really need to scroll the underlying page to the left, before opening the dialog to see the behavior I and @raidokaldma are talking about :)

**Comment by CyborgSemon** [comment]
ohhhhhhhhhhhh I see now! Sorry I didn't see it earlier. I was scrolling back to the original position before opening the dialog. I'll do some testing and get back to you.

**Comment by CyborgSemon** [comment]
Made a PR with a fix.
Thanks @raidokaldma for the suggestion! Good spotting!
